### ISSUE TITLE
NATS jetstream super cluster duplicates the messages

### ISSUE THREAD

**Issue by amalofeev-voc** [nats-io/k8s#497]
I'm deploying to k8s `NATS jetstream super cluster` using `helm` (3 kubernetes clusters with 3 NATS jetstream replicas). Gateway connections go through the addresses of load balancers (not a list of all addresses to servers/pods). Super cluster deploys successfully. However, when our application starts working with streams, we receive messages in a triple amount. If I connect to each cluster (`nats sub ">"`), then I don't see any duplications.
There is no problem if I deploy a super cluster on several linux virtual machines using docker. There is also no problem if the entire super cluster (with docker compose) is located, for example, locally.

Helm command:
```sh
for ctx in c1 c2 c3; do
  helm --kube-context $ctx install nats nats/nats -f k8s-nats-js-super-cluster.yaml --set=nats.jetstream.domain=$ctx --set=gateway.name=$ctx --set=cluster.name=$ctx
done
```

`k8s-nats-js-super-cluster.yaml`
```sh
---
nats:

  externalAccess: true

  logging:
    debug: true
    trace: true

  jetstream:
    enabled: true
    fileStorage:
      enabled: false
      storageDirectory: /data
    memStorage:
      enabled: true
      size: 100Mi

cluster:
  enabled: true
  listen: 0.0.0.0:6222
  replicas: 3
  routes:
    - nats-route://nats:6222

gateway:
  enabled: true
  listen: 0.0.0.0:7522

  gateways:
  - name: c1
    url: nats://c1.url:7522
  - name: c2
    url: nats://c2.url:7522
  - name: c3
    url: nats://c3.url:7522

natsbox:
  enabled: false
```
After deployment configs (0-index pods):
[c3-conf.txt](https://github.com/nats-io/k8s/files/8557584/c3-conf.txt)
[c2-conf.txt](https://github.com/nats-io/k8s/files/8557585/c2-conf.txt)
[c1-conf.txt](https://github.com/nats-io/k8s/files/8557587/c1-conf.txt)

Our application connects to an array of NATS addresses:
```sh
export NATS_URL="nats://c1.url:4222,nats://c2.url:4222,nats://c3.url:4222"
```

**Comment by caleblloyd** [comment]
@tbeets comment from Slack:

> Multiple JS Domains are not supported within one Supercluster.  If you have more than one JS Domain then you need to use Leaf Node connections rather than Gateway connection.

> Within one supercluster the Domain's RAFT META group spans all the servers in the supercluster -- all must have unique server names.  I am guessing you are seeing "placement" of one stream happening on three different servers of the same name.  They are all trying to ingest the same subject in the supercluster, etc.

**Comment by amalofeev-voc** [comment]
Thank you 

- JS Domain removed
- server names for entire super cluster are unique
```sh
for ctx in c1 c2 c3; do
  helm --kube-context $ctx upgrade --install -n voc nats-$ctx nats/nats -f eng-js-nats-gw-cluster.yaml --set=gateway.name=$ctx --set=cluster.name=$ctx
done
```

**Comment by wallyqs** [comment]
@amalofeev-voc could you share the version of the nats server where you ran into this? This might have been addressed in nats-server 2.8.0 (https://github.com/nats-io/nats-server/pull/2923.)
